### PR TITLE
add --ssl-peer-fingerprint for knife winrm fixing #298

### PR DESCRIPTION
**Pull request by hh**
https://github.com/WinRb/WinRM/pull/170 got us to where we need to be.

Thank @mwrock !

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,46 @@ without SSL there are limited options for ensuring the privacy of the
 plaintext transport. See the section on [Platform authentication
 support](#platform-winrm-authentication-support).
 
+If your cloud provides you with an rdp ssl fingerprint, you can
```
Comment (mwrock) [inline README.md]: I'd prefer to keep the instructions very minimal. Just a couple sentences stating if you know of a valid fingerprint that exists on the node, specify that as the `--ssl_peer_fingerprint` to be validated. This must reside in the certificate store on the node and there must be an active HTTPS winrm listener created with that certificate.

My reasoning here is:
* the readme is already quite lengthy
* it includes instructions on setting up SSL
* For testing purposes, now that both linux and windows workstations can use NTLM encryption over HTTP, thats the easiest route people are likely to pursue.

Does that sound reasonable?

Comment (mwrock) [inline README.md]: oh I just noticed #346. Would you mind replacing this documentation with the snippet from that PR? closing that one so its all in one pr?

File: lib/chef/knife/winrm_base.rb [hunk 2]
```diff
@@ -96,6 +96,10 @@ def self.included(includer)
             :default => :verify_peer,
             :proc => Proc.new { |verify_mode| verify_mode.to_sym }
 
+          option :ssl_peer_fingerprint,
```
Comment (mwrock) [inline lib/chef/knife/winrm_base.rb]: make sure to add `:ssl_peer_fingerprint` to https://github.com/chef/knife-windows/blob/master/spec/unit/knife/bootstrap_options_spec.rb#L104 which should get most of the tests here green

Comment (hh) [inline lib/chef/knife/winrm_base.rb]: Done!

File: spec/unit/knife/bootstrap_options_spec.rb [hunk 3]
```diff
@@ -92,6 +92,7 @@ def compare_property(sym)
       :forward_agent,
       :preserve_home,
       :ssh_gateway,
+      :ssl_peer_fingerprint,
```
Comment (mwrock) [inline spec/unit/knife/bootstrap_options_spec.rb]: looks like this is in `ref_ignore` but should be in `win_ignore`

Comment (hh) [inline spec/unit/knife/bootstrap_options_spec.rb]: Moved over

File: README.md (hunk 1 above)
Comment (hh) [inline README.md]: DONE! ;)

**Comment by hh** [comment]
@mwrock I think we are good for merge here. Can I get a review?

**Comment by mwrock** [comment]
:+1:

**Comment by mwrock** [comment]
I may make some very small edits to the readme, but overall this looks real good.

**Comment by mwrock** [comment]
cc @chef/client-windows

**Comment by hh** [comment]
@mwrock @chef/client-windows I rebased on top of master to make the merge easy.

**Comment by mwrock** [comment]
closing. will be merged with #350

**Pull request closed without merging**